frontend/lib/sdk/user.dart

```diff
@@ -0,0 +1,139 @@
+import 'package:dio/dio.dart';
+import 'package:dio_cookie_manager/dio_cookie_manager.dart';
+import 'package:cookie_jar/cookie_jar.dart';
+
+class User {
+  final String uid;
+
+  String email;
+  String? password;
+
+  User({
+    required this.email,
+    required this.uid,
+  });
+}
+
+class UserAPI {
+  Dio dio;
+
+  UserAPI({required this.dio}) {
+    dio.interceptors.add(CookieManager(CookieJar()));
+  }
+
+  final String loginUrl = "auth/login";
+  final String registerUrl = "auth/register";
+  final String logoutUrl = "auth/logout";
+  final String patchUrl = "user/";
+  final String meUrl = "user/me";
+
+  Future<void> register(String email, String password) async {
+    try {
+      Response response = await dio.post(registerUrl, data: {
+        "email": email,
+        "password": password,
+      });
+
+      if (response.statusCode != 201) {
+        throw Exception("Error registering");
+      }
+    } catch (e) {
+      rethrow;
+    }
+  }
+
+  Future<void> login(String email, String password) async {
+    try {
+      Response response = await dio.post(loginUrl, data: {
+        "email": email,
+        "password": password,
+      });
+
+      if (response.statusCode != 200) {
+        throw Exception("Error logging in");
+      }
+    } catch (e) {
+      rethrow;
+    }
+  }
+
+  Future<void> logout() async {
+    try {
+      Response response = await dio.post(logoutUrl);
+
+      if (response.statusCode != 200) {
+        throw Exception("Error logging out");
```

Comment:
Would be great to also log error to understand what happens in case of error?

Reply:
Those exception are throw and will be later catch and display to the user thanks to GetX when those functions will be called, a first view of how those throw are displayed can be seen in other branches.